src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,185 @@
+            product_name = winreg.QueryValueEx(daqmx_reg_key, "ProductName")[0]
+            product_version = winreg.QueryValueEx(daqmx_reg_key, "Version")[0]
+
+        if product_name == "NI-DAQmx":
+            return product_version
+    except Exception as e:
+        print(f"Info: No existing NI-DAQmx installation found")
+
+    return False
+
+
+def _get_temp_file_name():
+    """
+    Generate a temporary file name.
+    """
+    try:
+        _, temp_file_path = tempfile.mkstemp()
+        return temp_file_path + "_daq-mx-online.exe"
+    except Exception as e:
+        print(f"error: Unable to create temporary file: {e}")
+        return None
+
+
+def _get_url_and_version_for_windows_installer(metadata_url):
+    """
+    Parse the JSON data and retrieve the download link and version information.
+    """
+    print("Log1")
+    try:
+        with urllib.request.urlopen(metadata_url) as url:
+            data = json.load(url)
+            windows_os_meta = data.get("Windows", [])
+
+        for metadata in windows_os_meta:
+            location = metadata.get("Location")
+            version = metadata.get("Version")
+            if location and version:
+                return location, version
+
+    except Exception as e:
+        print(f"Unable to get download url: {e}")
+    return None, None
+
+
+def _install_daqmx(download_url, temp_file):
+    """
+    Download and launch NI-DAQmx Driver installation in an interactive mode
+    """
+    try:
+        urllib.request.urlretrieve(download_url, temp_file)
+        subprocess.Popen([temp_file], shell=True)
+    except Exception as e:
+        print(f"Error installing NI-DAQmx: {e}")
+
+
+def _ask_user_confirmation(question):
+    """
+    Prompt for user confirmation
+    """
+    while True:
+        response = input(question + " (yes/no): ").strip().lower()
+        if response in ["yes", "y"]:
+            return True
+        elif response in ["no", "n"]:
+            return False
+        else:
+            print("Please enter 'yes' or 'no'.")
+
+
+def install_daqmx_software():
+    """
+    Get the platform name on which the Python script is being executed.
+    Check for the existing installaion of NI-DAQmx Driver.
+    If NI-DAQmx driver is already installed it will prompt for an upgrade if applicable.
+    Download and launch NI-DAQmx Driver installation in an interactive mode.
+    """
+
+    if sys.platform.startswith("win"):
+        if _is_admin():
+            daqmx_latest_url, daqmx_latest_version = _get_url_and_version_for_windows_installer(
+                _META_DATA_URL
+            )
+            if daqmx_latest_version == None or daqmx_latest_url == None:
+                return -1
+            daqmx_installed_version = _check_daqmx_is_installed()
+            download_url, temp_file = daqmx_latest_url, _get_temp_file_name()
+            print(download_url)
+            print(temp_file)
+            if temp_file == None:
+               return -1
```

Comment:
Lots of cleanup to do here for "sad path" (opposite of [happy path](https://en.wikipedia.org/wiki/Happy_path)).

Reply:
I have refactored the code, @zhindes could you please check once ?